client/src/views/devices/GreetDeviceModal.vue

```diff
@@ -6,33 +6,52 @@
     :class="GreetDeviceStep[pageStep]"
   >
     <ms-wizard-stepper
-      v-show="pageStep > GreetDeviceStep.WaitForGuest && pageStep < GreetDeviceStep.Summary"
+      v-show="pageStep > GreetDeviceStep.WaitForGuest && pageStep < GreetDeviceStep.Summary && isLargeDisplay"
       :current-index="getStepperIndex()"
       :titles="['DevicesPage.greet.steps.hostCode', 'DevicesPage.greet.steps.guestCode']"
     />
     <ion-button
       slot="icon-only"
       @click="cancelModal()"
-      v-show="pageStep !== GreetDeviceStep.Summary"
       class="closeBtn"
+      v-if="pageStep !== GreetDeviceStep.Summary && isLargeDisplay"
     >
       <ion-icon
         :icon="close"
         class="closeBtn__icon"
       />
     </ion-button>
     <div class="modal">
-      <ion-header class="modal-header">
+      <ion-header
+        class="modal-header"
+        v-if="isLargeDisplay"
+      >
         <ion-title class="modal-header__title title-h2">
-          {{ $msTranslate(getTitleAndSubtitle().title) }}
+          {{ $msTranslate(getStep()[pageStep - 1]?.title) }}
         </ion-title>
         <ion-text
           class="modal-header__text body"
-          v-show="getTitleAndSubtitle().subtitle"
+          v-if="getStep()[pageStep - 1]?.subtitle"
         >
-          {{ $msTranslate(getTitleAndSubtitle().subtitle) }}
+          {{ $msTranslate(getStep()[pageStep - 1]?.subtitle) }}
         </ion-text>
       </ion-header>
+      <template v-else>
+        <small-display-modal-header
+          v-if="pageStep === GreetDeviceStep.WaitForGuest || pageStep === GreetDeviceStep.Summary"
+          @close-clicked="cancelModal()"
+          :hide-close-button="pageStep === GreetDeviceStep.Summary"
+          :title="getStep()[pageStep - 1]?.title"
+        />
+        <small-display-step-modal-header
+          v-else
+          @close-clicked="cancelModal()"
+          :title="'DevicesPage.addDevice'"
+          :icon="phonePortrait"
+          :steps="getStep()"
```

Comment:
Use a slice() to only pass the steps that the SmallDisplayStepModalHeader cares about. This way it doesn't have to do a `-2` that could change in some cases.
`:steps="steps.slice(1, steps.length - 1)"`